apps/costcontrol/test/unit/cost_control_test.js

```diff
@@ -498,4 +498,60 @@ suite('Cost Control Service Hub Suite >', function() {
       });
     }
   );
+
+  test(
+    'Each data cache is refreshed depending of the request type',
+    function(done) {
+      var expectedLastDataUsage = {
+        mobile: { total: 4800543137 },
+        wifi: { total: 112123944 }
+      };
+
+      var fakeSettings = { lastDataUsage: expectedLastDataUsage };
+
+      sinon.stub(SimManager, 'requestDataSimIcc', function (callback) {
+        (typeof callback === 'function') && callback({iccId:'12345'});
+      });
+
+      window.ConfigManager = new MockConfigManager({
+        fakeSettings: fakeSettings
+      });
+
+      CostControl.getInstance(function(service) {
+        service.request({type: 'datausage'}, function(result) {
+          assert.equal(Object.keys(result.data.mobile.apps).length, 0);
+          assert.equal(service.lastDataResults.mobile.total,
```

Comment:
When comparing, maybe you could create a couple of objects with the parts you want to compare and `deepEqual` those objects. You could do this in a custom `assertSomething` function.